runtime/libia2/exit_callgates_x86_64.c

```diff
@@ -0,0 +1,121 @@
+#include "ia2.h"
+
+#if defined(__x86_64__)
+
+// Exit compartment is hardcoded to pkey 1 (where libc/ld.so live)
+#define IA2_EXIT_COMPARTMENT_PKEY 1
+
+// PKRU mask for the exit compartment: allow access to pkey 0 and pkey 1 only.
+// PKRU uses 2 bits per protection key (Access Disable, Write Disable).
+// Result: 0xfffffff0 (pkey 0 and 1 accessible, pkeys 2-15 blocked)
+#define IA2_EXIT_PKRU (~((3U << (2 * IA2_EXIT_COMPARTMENT_PKEY)) | 3U))
+
+// Stringify helper macros for converting macro values to assembly immediates
+#define STR(x) #x
+#define XSTR(x) STR(x)
+
+__asm__(
+    ".text\n"
+    ".p2align 4,,15\n"
```

Comment:
I don't think the asm code here needs to be aligned. You can probably remove it or you should add a comment explaining why this is here.